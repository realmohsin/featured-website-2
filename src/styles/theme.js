import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
