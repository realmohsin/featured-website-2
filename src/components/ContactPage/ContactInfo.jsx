import React, { useState, useEffect } from 'react'
import clsx from 'clsx'
import { makeStyles, useTheme } from '@material-ui/core/styles'
import useMediaQuery from '@material-ui/core/useMediaQuery'
import { Container, Grid, Typography } from '@material-ui/core'
import { FaPhone } from 'react-icons/fa'

const useStyles = makeStyles(theme => ({
  contactSection: {
    padding: '4rem 0 10rem',
    [theme.breakpoints.down('md')]: {
      padding: '0rem 0 10rem'
    }
  },
  leftSide: {
    paddingLeft: '11rem',
    [theme.breakpoints.down('md')]: {
      textAlign: 'center',
      paddingLeft: 0
    }
  },
  rightSide: {
    paddingLeft: '10rem',
    [theme.breakpoints.down('md')]: {
      textAlign: 'center',
      paddingLeft: 0
    }
  },
  // contactTitle: {
  //   ...theme.custom.title2,
  //   marginBottom: '4rem'
  // },
  contactTitle: {
    ...theme.custom.title2,
    textAlign: 'left',
    fontSize: '5.6rem',
    paddingLeft: '11rem',
    marginBottom: '4rem',
    marginTop: '3rem',
    [theme.breakpoints.down('md')]: {
      fontSize: '4rem',
      paddingLeft: 0,
      textAlign: 'center',
      marginTop: 0
    },
    [theme.breakpoints.down('xs')]: {
      fontSize: '3rem'
    }
  },
  subsectionTitle: {
    textTransform: 'uppercase',
    letterSpacing: '1px',
    fontSize: '1.4rem',
    marginBottom: '1.4rem',
    color: theme.palette.secondary.main
  },
  daysTitle: {
    color: theme.palette.secondary.main,
    fontSize: '2rem',
    [theme.breakpoints.down('xs')]: {
      fontSize: '1.8rem'
    }
  },
  timesAndAddress: {
    color: theme.palette.primary.main,
    fontSize: '2.4rem',
    marginBottom: '5rem',
    lineHeight: '1.4',
    [theme.breakpoints.down('md')]: {
      marginBottom: '2rem'
    },
    [theme.breakpoints.down('xs')]: {
      fontSize: '2.1rem'
    }
  },
  telephoneNumber: {
    width: 'max-content',
    padding: '0.3rem 2.5rem',
    border: `2px solid ${theme.palette.secondary.main}`,
    fontSize: '3rem',
    marginBottom: '6rem',
    color: theme.palette.primary.main,
    letterSpacing: '0.7px',
    transition: '0.3s all',
    '&:hover': {
      color: theme.palette.primary.dark
    },
    [theme.breakpoints.down('md')]: {
      margin: '6rem auto 2rem'
    },
    [theme.breakpoints.down('xs')]: {
      fontSize: '2.5rem'
    }
  },
  phoneIcon: {
    marginRight: '1.2rem',
    transform: 'translateY(5px)'
  },
  contactMessage: {
    color: theme.palette.primary.main,
    fontSize: '1.9rem',
    [theme.breakpoints.down('md')]: {
      marginBottom: '10rem'
    }
  },
  outsideLink: {
    display: 'inline',
    color: theme.palette.secondary.main,
    margin: '0 0.3rem'
  },
  addressSubtitle: {
    fontSize: '1.6rem',
    color: theme.palette.primary.main,
    marginTop: '-3.6rem',
    marginBottom: '8rem',
    [theme.breakpoints.down('md')]: {
      marginTop: '-1rem'
    }
  },
  subwayStopLi: {
    display: 'flex',
    alignItems: 'center',
    marginBottom: '1.3rem',
    [theme.breakpoints.down('md')]: {
      marginLeft: '25%'
    },
    [theme.breakpoints.down('xs')]: {
      marginLeft: '5%'
    }
  },
  subwayDirections: {
    fontSize: '2.3rem',
    color: theme.palette.primary.main,
    marginBottom: '3rem',
    [theme.breakpoints.down('xs')]: {
      fontSize: '1.6rem'
    }
  },
  subwayStopText: {
    color: theme.palette.primary.main,
    marginLeft: '1.8rem',
    fontSize: '2rem',
    [theme.breakpoints.down('xs')]: {
      fontSize: '1.6rem'
    }
  },
  singleTrainText: {
    marginLeft: '6.2rem'
  },
  subwayIcon: {
    borderRadius: '50%',
    margin: '0 0.5rem',
    color: 'white',
    fontWeight: 'bold',
    width: '3.4rem',
    height: '3.4rem',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: '2rem'
  }
}))

const SubwayIcon = ({ symbol, color }) => {
  const classes = useStyles()
  return (
    <span className={classes.subwayIcon} style={{ backgroundColor: color }}>
      {symbol}
    </span>
  )
}

const ContactSection = props => {
  const theme = useTheme()
  const classes = useStyles()

  return (
    <section className={classes.contactSection}>
      <Container>
        <h1 className={classes.contactTitle}>Contact Romano Law</h1>
        <Grid container>
          <Grid item xs={12} md={6} className={classes.leftSide}>
            <Typography variant='body1' className={classes.subsectionTitle}>
              Hours
            </Typography>
            <Typography variant='body1' className={classes.daysTitle}>
              Mondays through Friday:
            </Typography>
            <Typography variant='body1' className={classes.timesAndAddress}>
              9:00AM - 7:00PM
            </Typography>
            <Typography variant='body1' className={classes.daysTitle}>
              Saturday:
            </Typography>
            <Typography variant='body1' className={classes.timesAndAddress}>
              9:00AM - 7:00PM
            </Typography>
            <a className={classes.telephoneNumber}>
              <FaPhone
                color={theme.palette.secondary.main}
                className={classes.phoneIcon}
              />
              (212) 865 - 9848
            </a>
            <Typography variant='body1' className={classes.contactMessage}>
              Click
              <a href='https://calendly.com/' className={classes.outsideLink}>
                HERE
              </a>
              to schedule a complimentary call.
            </Typography>
          </Grid>
          <Grid item xs={12} md={6} className={classes.rightSide}>
            <Typography variant='body1' className={classes.subsectionTitle}>
              Address
            </Typography>
            <Typography variant='body1' className={classes.timesAndAddress}>
              Romano Law PLLC <br />
              55 Broad Street, 18th Fl. <br />
              New York, NY 10004
            </Typography>
            <Typography variant='body1' className={classes.addressSubtitle}>
              For directions click{' '}
              <a href='https://google.com/maps' className={classes.outsideLink}>
                HERE
              </a>
            </Typography>
            <Typography variant='body1' className={classes.subsectionTitle}>
              Subway Stops
            </Typography>
            <Typography variant='body1' className={classes.subwayDirections}>
              We are located in the Financial District of New York City, two
              blocks south of Wall Street.
            </Typography>
            <ul className={classes.subwayStopsUL}>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#ee352e' symbol={2} />
                <SubwayIcon color='#ee352e' symbol={3} />
                <Typography variant='body1' className={classes.subwayStopText}>
                  Wall St.
                </Typography>
              </li>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#00933c' symbol={4} />
                <SubwayIcon color='#00933c' symbol={5} />
                <Typography variant='body1' className={classes.subwayStopText}>
                  Bowling Green (or Wall)
                </Typography>
              </li>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#996633' symbol={'J'} />
                <SubwayIcon color='#996633' symbol={'Z'} />
                <Typography variant='body1' className={classes.subwayStopText}>
                  Broad St.
                </Typography>
              </li>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#fccc0a' symbol={'R'} />
                <SubwayIcon color='#fccc0a' symbol={'W'} />
                <Typography variant='body1' className={classes.subwayStopText}>
                  Rector St. (or Whitehall)
                </Typography>
              </li>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#ee352e' symbol={1} />
                <Typography
                  variant='body1'
                  className={clsx(
                    classes.subwayStopText,
                    classes.singleTrainText
                  )}
                >
                  Rector St. (or South Ferry)
                </Typography>
              </li>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#0039a6' symbol={'A'} />
                <SubwayIcon color='#0039a6' symbol={'C'} />
                <Typography variant='body1' className={classes.subwayStopText}>
                  Fulton St.
                </Typography>
              </li>
              <li className={classes.subwayStopLi}>
                <SubwayIcon color='#0039a6' symbol={'E'} />

                <Typography
                  variant='body1'
                  className={clsx(
                    classes.subwayStopText,
                    classes.singleTrainText
                  )}
                >
                  World Trade Center
                </Typography>
              </li>
            </ul>
          </Grid>
        </Grid>
      </Container>
    </section>
  )
}

export default ContactSection
