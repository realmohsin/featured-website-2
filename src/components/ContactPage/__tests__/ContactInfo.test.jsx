import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import { createMuiTheme } from '@material-ui/core/styles';
import ContactInfo from '../ContactInfo';

describe('ContactInfo', () => {
  const theme = createMuiTheme({
  custom: {
    title2: {
      fontSize: '2.8rem',
      lineHeight: 1.2,
      fontWeight: 600,
      marginBottom: '1.5rem',
    },
    title5: {
      fontSize: '1.8rem',
      fontWeight: 500,
      marginBottom: '1rem',
    },
    body1: {
      fontSize: '1.6rem',
      lineHeight: 1.6,
    },
  },
  palette: {
    primary: {
      main: '#000000',
    },
    secondary: {
      main: '#ffffff',
    },
  },
  breakpoints: {
    down: () => '@media (max-width: 600px)',
  },
});

  const renderWithTheme = (component) => {
    return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
  };

  it('renders contact information section', () => {
    renderWithTheme(<ContactInfo />);
    
    // Check section title
    expect(screen.getByText(/Contact Romano Law/i)).toBeInTheDocument();
    
    // Check section headers
    expect(screen.getByText(/Hours/i)).toBeInTheDocument();
    expect(screen.getByText(/Address/i)).toBeInTheDocument();
    
    // Check phone number
    const phoneLink = screen.getByRole('link', { name: /\(212\) 865 - 9848/i });
    expect(phoneLink).toBeInTheDocument();
    expect(phoneLink).toHaveAttribute('href', 'tel:+12128659848');
    
    // Check address
    expect(screen.getByText(/55 Broad Street, 18th Fl\./i)).toBeInTheDocument();
    expect(screen.getByText(/New York, NY 10004/i)).toBeInTheDocument();
    
    // Check subway stops section
    expect(screen.getByText(/Subway Stops/i)).toBeInTheDocument();
    expect(screen.getByText(/We are located in the Financial District of New York City/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = renderWithTheme(<ContactInfo />);
    expect(container).toMatchSnapshot();
  });
});
