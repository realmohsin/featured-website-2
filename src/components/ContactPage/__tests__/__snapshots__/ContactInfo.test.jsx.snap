// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfo matches snapshot 1`] = `
<div>
  <section
    class="makeStyles-contactSection-18"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <h1
        class="makeStyles-contactTitle-21"
      >
        Contact Romano Law
      </h1>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root makeStyles-leftSide-19 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <p
            class="MuiTypography-root makeStyles-subsectionTitle-22 MuiTypography-body1"
          >
            Hours
          </p>
          <p
            class="MuiTypography-root makeStyles-daysTitle-23 MuiTypography-body1"
          >
            Mondays through Friday:
          </p>
          <p
            class="MuiTypography-root makeStyles-timesAndAddress-24 MuiTypography-body1"
          >
            9:00AM - 7:00PM
          </p>
          <p
            class="MuiTypography-root makeStyles-daysTitle-23 MuiTypography-body1"
          >
            Saturday:
          </p>
          <p
            class="MuiTypography-root makeStyles-timesAndAddress-24 MuiTypography-body1"
          >
            9:00AM - 7:00PM
          </p>
          <a
            class="makeStyles-telephoneNumber-25"
            href="tel:+12128659848"
          >
            <svg
              class="makeStyles-phoneIcon-26"
              color="#CCAB63"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(204, 171, 99);"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
              />
            </svg>
            (212) 865 - 9848
          </a>
          <p
            class="MuiTypography-root makeStyles-contactMessage-27 MuiTypography-body1"
          >
            Click
            <a
              class="makeStyles-outsideLink-28"
              href="https://calendly.com/"
            >
              HERE
            </a>
            to schedule a complimentary call.
          </p>
        </div>
        <div
          class="MuiGrid-root makeStyles-rightSide-20 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
        >
          <p
            class="MuiTypography-root makeStyles-subsectionTitle-22 MuiTypography-body1"
          >
            Address
          </p>
          <p
            class="MuiTypography-root makeStyles-timesAndAddress-24 MuiTypography-body1"
          >
            Romano Law PLLC 
            <br />
            55 Broad Street, 18th Fl. 
            <br />
            New York, NY 10004
          </p>
          <p
            class="MuiTypography-root makeStyles-addressSubtitle-29 MuiTypography-body1"
          >
            For directions click
             
            <a
              class="makeStyles-outsideLink-28"
              href="https://google.com/maps"
            >
              HERE
            </a>
          </p>
          <p
            class="MuiTypography-root makeStyles-subsectionTitle-22 MuiTypography-body1"
          >
            Subway Stops
          </p>
          <p
            class="MuiTypography-root makeStyles-subwayDirections-31 MuiTypography-body1"
          >
            We are located in the Financial District of New York City, two blocks south of Wall Street.
          </p>
          <ul>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(238, 53, 46);"
              >
                2
              </span>
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(238, 53, 46);"
              >
                3
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 MuiTypography-body1"
              >
                Wall St.
              </p>
            </li>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(0, 147, 60);"
              >
                4
              </span>
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(0, 147, 60);"
              >
                5
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 MuiTypography-body1"
              >
                Bowling Green (or Wall)
              </p>
            </li>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(153, 102, 51);"
              >
                J
              </span>
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(153, 102, 51);"
              >
                Z
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 MuiTypography-body1"
              >
                Broad St.
              </p>
            </li>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(252, 204, 10);"
              >
                R
              </span>
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(252, 204, 10);"
              >
                W
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 MuiTypography-body1"
              >
                Rector St. (or Whitehall)
              </p>
            </li>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(238, 53, 46);"
              >
                1
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 makeStyles-singleTrainText-33 MuiTypography-body1"
              >
                Rector St. (or South Ferry)
              </p>
            </li>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(0, 57, 166);"
              >
                A
              </span>
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(0, 57, 166);"
              >
                C
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 MuiTypography-body1"
              >
                Fulton St.
              </p>
            </li>
            <li
              class="makeStyles-subwayStopLi-30"
            >
              <span
                class="makeStyles-subwayIcon-34"
                style="background-color: rgb(0, 57, 166);"
              >
                E
              </span>
              <p
                class="MuiTypography-root makeStyles-subwayStopText-32 makeStyles-singleTrainText-33 MuiTypography-body1"
              >
                World Trade Center
              </p>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </section>
</div>
`;
